### PR TITLE
Hotfix AI not using all goals

### PR DESCRIPTION
**Pull request by someaddons**
# Changes proposed in this pull request:
- Goals which have no mutex flags can tick again.

Review please

### CODE REVIEW

File: src/api/java/com/minecolonies/api/entity/CustomGoalSelector.java [hunk 1]
```diff
@@ -77,6 +72,7 @@ public CustomGoalSelector(@NotNull final GoalSelector old)
         super(old.profiler);
         importFrom(old);
         super.goals = this.goals;
+        super.profiler = this.profiler;
```
Comment (Raycoms) [inline src/api/java/com/minecolonies/api/entity/CustomGoalSelector.java:75]: do we need the this.profiler at all?

File: src/api/java/com/minecolonies/api/entity/CustomGoalSelector.java [hunk 2]
```diff
@@ -89,7 +85,7 @@ public CustomGoalSelector(@NotNull final IProfiler profiler)
         super(profiler);
         this.profiler = profiler;
         super.goals = this.goals;
-
+        super.profiler = this.profiler;
```
Comment (Raycoms) [inline src/api/java/com/minecolonies/api/entity/CustomGoalSelector.java:88]: twice?

Comment (someaddons) [reply src/api/java/com/minecolonies/api/entity/CustomGoalSelector.java:88]: ye for each constructor

Comment (Raycoms) [reply src/api/java/com/minecolonies/api/entity/CustomGoalSelector.java:88]: Ahh okay, didn't see it was the next constructor in this small review window =D

File: src/api/java/com/minecolonies/api/entity/CustomGoalSelector.java (hunk 1 above)
Comment (someaddons) [reply src/api/java/com/minecolonies/api/entity/CustomGoalSelector.java:75]: ye its used to set the minecraft debug info, and assigned to super incase anything tries to access it on the base minecraft type.

**Pull request merged**